cumulus/parachains/integration-tests/emulated/tests/collectives/collectives-westend/src/tests/proxy.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use asset_hub_westend_runtime::{
+	xcm_config::LocationToAccountId as AssetHubLocationToAccountId, ProxyType as AssetHubProxyType,
+};
+use codec::Encode;
+use collectives_westend_runtime::ProxyType as CollectivesProxyType;
+use frame_support::{
+	assert_ok,
+	traits::fungible::{Inspect, Mutate},
+};
+use xcm_executor::traits::ConvertLocation;
+
+#[test]
+fn cross_chain_pure_proxy() {
+	// proxy of the pure account.
+	let alice = Westend::account_id_of("Alice");
+	// spawner of the pure account, delegates to the pure account to Alice.
+	let bob = Westend::account_id_of("Bob");
+
+	// Bob creates a pure account on Collectives Chain.
+	let pure = CollectivesWestend::execute_with(|| {
+		type Proxy = <CollectivesWestend as CollectivesWestendPallet>::Proxy;
+		type RuntimeOrigin = <CollectivesWestend as Chain>::RuntimeOrigin;
+		type RuntimeEvent = <CollectivesWestend as Chain>::RuntimeEvent;
+
+		assert_ok!(Proxy::create_pure(
+			RuntimeOrigin::signed(bob.clone()),
+			CollectivesProxyType::Any,
+			0,
+			0
+		));
+
+		let events = CollectivesWestend::events();
+		match events.iter().last() {
+			Some(RuntimeEvent::Proxy(pallet_proxy::Event::PureCreated { pure, .. })) =>
+				pure.clone(),
+			_ => panic!("Expected PureCreated event"),
+		}
+	});
+
+	// Bob adds Alice as a proxy to the pure account on Collectives Chain.
+	CollectivesWestend::execute_with(|| {
+		type Proxy = <CollectivesWestend as CollectivesWestendPallet>::Proxy;
+		type Balances = <CollectivesWestend as CollectivesWestendPallet>::Balances;
+		type Runtime = <CollectivesWestend as Chain>::Runtime;
+		type RuntimeOrigin = <CollectivesWestend as Chain>::RuntimeOrigin;
+		type RuntimeEvent = <CollectivesWestend as Chain>::RuntimeEvent;
+		type RuntimeCall = <CollectivesWestend as Chain>::RuntimeCall;
+
+		// fund pure account for a deposit required for adding Alice as a proxy.
+		assert_ok!(<Balances as Mutate<_>>::mint_into(
+			&pure,
+			<Balances as Inspect<_>>::minimum_balance() * 100
+		));
+
+		let transfer_proxy_call = RuntimeCall::Proxy(pallet_proxy::Call::<Runtime>::add_proxy {
```

Comment:
This confused me reading it, since it's not transferring.
```suggestion
		let add_proxy_call = RuntimeCall::Proxy(pallet_proxy::Call::<Runtime>::add_proxy {
```